pkg/build/types/bundle.go

```diff
@@ -83,10 +112,116 @@ func NewBundle(bundleDir, bundlePrefix string) (b *Bundle, err error) {
 		"rootfs": "fs",
 	}
 
+	var input = "Default"
+	if encrypt == true {
```

Comment:
this branch is too long. Please move it to a separate function.

I was thinking it should be possible to set up the contents of the file system as usual, create the squashfs filesystem as usual, and then create a loop device of the correct size (squashfs + ~ 16 MB overhead, exact number is 16384 for the metadata + 16744448 key slots, it might be possible to reduce the size of the key slot storage), format it and the copy the squashfs data into the encrypted volume (e.g. `dd if=squashfs.img of=/dev/mapper/luks-volume ...`).